hugr-model/src/v0/text/parse.rs

```diff
@@ -50,33 +52,59 @@ pub fn parse<'a>(input: &'a str, bump: &'a Bump) -> Result<ParsedModule<'a>, Par
 struct ParseContext<'a> {
     module: Module<'a>,
     bump: &'a Bump,
+    vars: VarTable<'a>,
+    links: LinkTable<&'a str>,
+    symbols: SymbolTable<'a>,
+    implicit_imports: FxHashMap<&'a str, NodeId>,
 }
 
 impl<'a> ParseContext<'a> {
     fn new(bump: &'a Bump) -> Self {
         Self {
             module: Module::default(),
+            symbols: SymbolTable::default(),
+            links: LinkTable::default(),
+            vars: VarTable::default(),
+            implicit_imports: FxHashMap::default(),
             bump,
         }
     }
 
     fn parse_module(&mut self, pair: Pair<'a, Rule>) -> ParseResult<()> {
         debug_assert_eq!(pair.as_rule(), Rule::module);
         let mut inner = pair.into_inner();
+
+        self.module.root = self.module.insert_region(Region::default());
+        self.symbols.enter(self.module.root);
+        self.links.enter(self.module.root);
+
+        // TODO: What scope does the metadata live in?
         let meta = self.parse_meta(&mut inner)?;
+        let explicit_children = self.parse_nodes(&mut inner)?;
 
-        let children = self.parse_nodes(&mut inner)?;
+        let mut children = BumpVec::with_capacity_in(
```

Comment:
Can't this be something like `from_slice(explicit_children.into_iter().chain(self.implicit_imports.drain().....))`?

Reply:
Afaik `chain` does not implement `ExactSizeIterator`...